### PR TITLE
Add management_group resources

### LINKED ISSUE #328: Please Convert azurerm_management_group resources to use the new generic resource
Currently, the azurerm_management_group and azurerm_management_groups resources have not been converted over to the new generic resource method.

This blocks their use with Azure China endpoint.

### PR DESCRIPTION
**Pull request by omerdemirok**
### Description

- Re-write management group(s) resources with the new backend
- Fix some typos
- Update integration tests

### Issues Resolved

Resolves #328 

### Check List

- [x] New functionality includes tests
- [ ] All tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: docs/resources/azure_management_group.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+```
+## Parameters
+
+| Name                           | Description                                                                       |
+|--------------------------------|-----------------------------------------------------------------------------------|
+| name         | Management group name. `20000000-0001-0000-0000-000000000000` |
+| group_id     | Alias for the `name` parameter. |
+| resource_id  | The unique resource ID. `/providers/Microsoft.Management/managementGroups/{groupId}` |
+| expand       | Optional. The `expand: 'children'` includes children in the response. `expand: 'path'` includes the path from the root group to the current group. |
+| recurse      | Optional. The `recurse: true` includes the entire hierarchy in the response. Note that `expand: 'children'` must be passed up if `recurse` is set to `true`. |
```
Comment (rmoles) [inline docs/resources/azure_management_group.md:48]: `must be passed up` is this saying that `expand: children` must also be set it `recurse` is set to true?

Comment (omerdemirok) [reply docs/resources/azure_management_group.md:48]: Yes. We can reword if it is not clear.

File: libraries/azure_management_group.rb [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    opts[:resource_uri] = "providers/#{opts[:resource_provider]}"
+    opts[:add_subscription_id] = false
+    opts[:resource_identifiers] = %i(group_id)
+    opts[:allowed_parameters] = %i(expand recurse filter)
+    # For backward compatibility.
+    opts[:query_parameters] = {
+      '$recurse' => opts[:recurse] || false,
+    }
+    opts[:query_parameters].merge!('$expand' => opts[:expand]) unless opts[:expand].nil?
+    # Note that $expand=children must be passed up if $recurse is set to true.
```
Comment (rmoles) [inline libraries/azure_management_group.rb:26]: This code answers my question... I think the wording should be changed slightly. `$expand=children must be set if $recurse is set to true.`
OR
`$expand=children will be set if $recurse is set to true.`

**Review by rmoles — APPROVED** [review]
Thanks @omerdemirok great work as always. Just a small comment regarding wording.

File: libraries/azure_management_group.rb (hunk 2 above)
Comment (rmoles) [reply libraries/azure_management_group.rb:26]: could this comment also be updated for consistency?

**Pull request merged**